app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java

```diff
@@ -65,16 +65,18 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
           }
         });
 
-    new ProcessJson().execute("");
+    // Try pulling data from GraphQL, if not fallback to json from cornell dining
+    NetworkUtilities.getEateries(presenter,this);
+    if (JSON_FALLBACK) {
+      new ProcessJson().execute("");
+    }
   }
 
   public class ProcessJson extends AsyncTask<String, Void, ArrayList<EateryBaseModel>> {
```

Comment:
Right now, half of the networking logic is on the activity and the other half is in a networking class. I feel like it's almost better to integrate `ProcessJson` into `NetworkUtils`.

Reply:
Discussed privately. Resolved